xarray_schema/dataarray.py

```diff
@@ -60,16 +62,16 @@ def __init__(
         attrs: Mapping[str, Any] = None,
         checks: List[Callable] = None,
     ) -> None:
-
-        self.dtype = dtype
-        self.shape = shape
-        self.dims = dims
-        self.name = name
-        self.coords = coords
-        self.chunks = chunks
-        self.attrs = attrs
-        self.array_type = array_type
-        self.checks = checks
+        # see https://github.com/python/mypy/issues/3004
```

Comment:
mypy seems to fail when it is applied to a property whose setter has a different return type than the parameter type of the corresponding getter.